internal/configs/virtualserver.go

```diff
@@ -342,6 +347,12 @@ func generateUpstream(upstreamName string, upstream conf_v1alpha1.Upstream, isEx
 			FailTimeout: generateString(upstream.FailTimeout, cfgParams.FailTimeout),
 			MaxConns:    generateIntFromPointer(upstream.MaxConns, cfgParams.MaxConns),
 			Resolve:     isExternalNameSvc,
+			SlowStart:   upstream.SlowStart,
+		}
+		if incompatibleLBMethods[upstream.LBMethod] {
```

Comment:
I would further suggest putting the logic for checking for incompatible method in a separate function and writing a unit test for that function.

```go
	lbMethod := generateLBMethod(upstream.LBMethod, cfgParams.LBMethod)
	slowStart := 	for _, e := range endpoints {
		s := version2.UpstreamServer{
			Address:     e,
			MaxFails:    generateIntFromPointer(upstream.MaxFails, cfgParams.MaxFails),
			FailTimeout: generateString(upstream.FailTimeout, cfgParams.FailTimeout),
			MaxConns:    generateIntFromPointer(upstream.MaxConns, cfgParams.MaxConns),
			Resolve:     isExternalNameSvc,
                         SlowStart:  generateSlowStart(lbMethod, upstream.SlowStart, isPlus)

,
		}
		upsServers = append(upsServers, s)
	}


	return version2.Upstream{
		Name:      upstreamName,
		Servers:   upsServers,
		LBMethod: lbMethod,
		Keepalive: generateIntFromPointer(upstream.Keepalive, cfgParams.Keepalive),
	}
```

Note that `isPlus` will have to be added to the list of `generateUpstream` arguments. We need it because slow start only works for NGINX Plus.

Additionally, also note that `hash` method is defined by  `hash` + some varialble(s), for example `$hash $request_id`. As a result, the check `incompatibleLBMethods[lbMethod]` for hash method won't work. We need to check for the prefix, not the exact match here. Please also see the implementation of ParseLBMethod function https://github.com/nginxinc/kubernetes-ingress/blob/master/internal/configs/parsing_helpers.go#L92

Also, I suggest renaming `incompatibleLBMethods ` to `incompatibleLBMethodsForSlowStart `and making it a global variable similar to https://github.com/nginxinc/kubernetes-ingress/blob/master/internal/configs/parsing_helpers.go#L119 so that the map is only created once.